packages/core/src/agent/getOutboundMessageContext.ts

```diff
@@ -297,8 +297,11 @@ async function addExchangeDataToMessage(
     associatedRecord: BaseRecordAny
   }
 ) {
+  const legacyInvitationMetadata = outOfBandRecord?.metadata.get(OutOfBandRecordMetadataKeys.LegacyInvitation)
+
   // Set the parentThreadId on the message from the oob invitation
-  if (outOfBandRecord) {
+  // If connectionless is used, we should not add the parentThreadId
+  if (outOfBandRecord && legacyInvitationMetadata?.legacyInvitationType !== 'connectionless') {
```

Comment:
Would it make more sense to have `legacyInvitationType` be an enum?

Reply:
Yeah probably :)

Reply:
In `ConnectionRecord` we have the `protocol` field where we define the handshake protocol for the DID Exchange, and in the case of DIDComm V2, where no handshake protocol is involved, we set it to `None`. If this `legacyInvitationType` refers to the message type, would it make sense to call it 'none'. in the case of a connection-less exchange?

Reply:
> would it make sense to call it 'none'. in the case of a connection-less exchange?

Well in that case we don't know explicitly if it's a connectionless message, and we have custom behaviour for that as you can see here. I'd still call a message with a `~service` an invitation to do something.

So invitation type here refers less to connections, and more to an invitation to an exchange that is sent not over didcomm. (connectionless, connection v1, oob v1, etc..)